cli/js/tests/unit_test_runner.ts

```diff
@@ -127,31 +127,50 @@ async function runTestsForPermissionSet(
   // Wait for worker subprocess to go online
   const conn = await listener.accept();
 
+  let err: Error | undefined = undefined;
   let expectedPassedTests;
   let endEvent;
 
-  for await (const line of readLines(conn)) {
-    const msg = JSON.parse(line);
-
-    if (msg.kind === Deno.TestEvent.Start) {
-      expectedPassedTests = msg.tests;
-      await reporter.start(msg);
-      continue;
-    } else if (msg.kind === Deno.TestEvent.TestStart) {
-      await reporter.testStart(msg);
-      continue;
-    } else if (msg.kind === Deno.TestEvent.TestEnd) {
-      await reporter.testEnd(msg);
-      continue;
-    } else {
-      endEvent = msg;
-      await reporter.end(msg);
-      break;
+  try {
+    for await (const line of readLines(conn)) {
+      const msg = JSON.parse(line);
+
+      if (msg.kind === Deno.TestEvent.Start) {
+        expectedPassedTests = msg.tests;
+        await reporter.start(msg);
+        continue;
+      } else if (msg.kind === Deno.TestEvent.TestStart) {
+        await reporter.testStart(msg);
+        continue;
+      } else if (msg.kind === Deno.TestEvent.TestEnd) {
+        await reporter.testEnd(msg);
+        continue;
+      } else {
+        endEvent = msg;
+        await reporter.end(msg);
+        break;
+      }
     }
+  } catch (e) {
+    err = e;
+  } finally {
+    // Close socket to worker, it should shutdown gracefully.
+    conn.close();
   }
 
-  // Close socket to worker, it should shutdown gracefully.
-  conn.close();
+  if (typeof err !== "undefined") {
```

Comment:
Is this necessary? Can't it just be `if (err instanceof Deno.errors.ConnectionReset) {` ?